### PR TITLE
feat: resource scan

### PR DESCRIPTION
**Pull request by AbhishekA1509**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Shared/constants.tsx [hunk 1]
```diff
@@ -37,3 +37,4 @@ export const ModuleNameMap = {
 // TODO: Convert to object later if more formats arise
 export const DATE_TIME_FORMAT_STRING = 'ddd DD MMM YYYY HH:mm:ss'
 export const API_TOKEN_PREFIX = 'API-TOKEN:'
+export const STRINGIFIED_TRUE = 'true'
```
Comment (eshankvaish) [inline src/Shared/constants.tsx:40]: .

File: src/Shared/Components/ScannedObjectBar/styles.scss [hunk 2]
```diff
@@ -0,0 +1,34 @@
+.scanned-object__bar {
```
Comment (eshankvaish) [inline src/Shared/Components/ScannedObjectBar/styles.scss:1]: pls use util classes

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -77,7 +77,8 @@
         "react-router": "^5.3.0",
         "react-router-dom": "^5.3.0",
         "react-draggable": "^4.4.5",
-        "yaml": "^2.4.1"
+        "yaml": "^2.4.1",
+        "dompurify": "^3.0.2"
     },
```
[the remark this answers is not in the document]
Comment (AbhishekA1509) [reply package.json:82]: Leaving as discussed

File: src/Shared/types.ts [hunk 4]
```diff
@@ -137,3 +137,49 @@ export interface MaterialSecurityInfoType {
...
+    VolumeSnapshot = 'VolumeSnapshot',
+    VolumeSnapshotContent = 'VolumeSnapshotContent',
+    VolumeSnapshotClass = 'VolumeSnapshotClass',
+    PodDisruptionBudget = 'PodDisruptionBudget',
+    Event = 'Event',
+    Namespace = 'Namespace',
+    Overview = 'Overview',
+}
+
+export type NodeType = keyof typeof Nodes
```
[the remark this answers is not in the document]
Comment (AbhishekA1509) [reply src/Shared/types.ts:185]: imported as is from dashboard

File: src/Shared/constants.tsx (hunk 1 above)
Comment (AbhishekA1509) [reply src/Shared/constants.tsx:40]: Done

File: src/Shared/Components/ScannedObjectBar/styles.scss (hunk 2 above)
Comment (AbhishekA1509) [reply src/Shared/Components/ScannedObjectBar/styles.scss:1]: imported from dashboard

File: src/Shared/types.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (eshankvaish) [reply src/Shared/types.ts:185]: Please sync with QA for testing this. This seems to be a potential bug

**Review by eshankvaish — APPROVED** [review]
pls add the respective todo

**Review by vivek-devtron — DISMISSED** [review]

**Pull request merged**